nilearn/plotting/displays.py

```diff
@@ -854,8 +859,9 @@ def _locator(self, axes, renderer):
 
         if self._colorbar:
             adjusted_width = self._colorbar_width/len(self.axes)
-            ticks_margin = adjusted_width*self._colorbar_labels_margin
-            x1 = x1 - (adjusted_width+ticks_margin)
+            right_margin = self._colorbar_margin['right'] / len(self.axes)
+            ticks_margin = self._colorbar_margin['left'] / len(self.axes)
+            x1 = x1 - (adjusted_width + right_margin + ticks_margin)
```

Comment:
Note: the _locator method seems very redundant across classes. maybe there will be something to refactor here in a future PR.